mm2src/coins/lp_coins.rs

```diff
@@ -699,6 +706,26 @@ pub trait SwapOps {
         Box::new(futures01::future::ok(result))
     }
 
+    /// Whether the receiver of a payment of this coin can fail/rollback/release the payment manually or not.
+    fn can_be_released(&self) -> bool { false }
+
+    /// Fails an HTLC back to its origin to free resources.
+    ///
+    /// # Important
+    ///
+    /// Taker shouldn't fail a swap htlc to the maker/origin until the taker swap payment can be refunded successfully.
+    fn fail_htlc_backwards(&self, _other_side_tx: &[u8]) -> FailHTLCFut<()> { Box::new(futures01::future::ok(())) }
+
+    /// Whether the swap payment is refunded automatically or not when the locktime expires, or the other side fails the HTLC.
+    fn is_auto_refundable(&self) -> bool { false }
+
+    /// Waits for an htlc to be refunded automatically and returns the refund transaction.
+    fn wait_for_htlc_refund(&self, _tx: &[u8], _locktime: u64) -> TransactionFut {
+        Box::new(futures01::future::err(TransactionErr::Plain(
+            "wait_for_htlc_refund is not supported for this coin!".into(),
+        )))
+    }
```

Comment:
Using `TransactionFut` here is quite tricky because no refund transaction is actually broadcast, at least for `TendermintCoin/Token`. Could you please try to find an alternative solution?

Reply:
>  because no refund transaction is actually broadcast, at least for TendermintCoin/Token

Same for lightning, the htlc just gets rolled back / removed from the commitment transactions. I think this is the case for any auto refund coin, so I will make `wait_for_htlc_refund` return `()`

Reply:
I added `Todo` on `is_auto_refundable`,`wait_for_htlc_refund` implementations for `TendermintCoin`,`TendermintToken`